swupd/packs.go

```diff
@@ -232,10 +242,27 @@ func WritePack(w io.Writer, fromManifest, toManifest *Manifest, outputDir, chroo
 			return nil, err
 		}
 
+		dManifest.Files = append(dManifest.Files, d.from)
 		info.DeltaCount++
 		hasDelta[d.to.Hash] = d
 	}
 
+	if fromManifest != nil {
+		dManifest.Header.FileCount = uint32(len(dManifest.Files))
+		if dManifest.Header.FileCount > 0 {
```

Comment:
I think it would be easier to read and you save an indentation level if you turn this into a guard instead of a nested if, so
```go
if dManifest.Header.FileCount <= 0 {
    continue
}
...
```
but ymmv

Reply:
This isn't inside a loop so I can't use continue. I don't see how I can reduce the number of nested ifs. Any suggestion?

Reply:
Do you actually reuse this object anywhere else other than here? If not, you don't really need to set `dManifest.Header.FileCount` in the line above when `len(dManifest.Files)` is 0 and you can merge both ifs into a single one.

Also note that this function is kinda big. It wouldn't be bad to put this into a secondary function so you are able to use a `return` statement.